.github/workflows/ci-test.yml

```diff
@@ -537,13 +533,10 @@ jobs:
           ref: ${{ github.event.pull_request.head.sha }}
           fetch-depth: 0 #get all the history!!!
       - uses: ./.github/actions/build-setup
-      - uses: actions/cache@v2.1.6
-        with:
-          path: "/opt/cargo/git\n/opt/cargo/registry\n/opt/cargo/.package-cache"
-          key: crates-${{ runner.os }}-${{ hashFiles('Cargo.lock') }}
-          restore-keys: "crates-${{ runner.os }}"
+      - uses: Swatinem/rust-cache@c5ed9ba6b7e1bb8aff90d43acd2f0af4990fa57c
       - name: run unit tests
         run: |
+          rm -rf ./target/x-scratch # Work around bad cached data
```

Comment:
I am curious, is this due to the workspace-hack in x?

Reply:
no not workspace-hack, but x's logic to run tests with `--chaged-since`. Funny enough theres a todo comment in the code that would have eliminated this buggy behavior. This line should really only been needed until a new entry is added to the cache (since I added cleanup logic for this in the build teardown action)